cornice/util.py

```diff
@@ -84,20 +96,35 @@ def to_list(obj):
     return obj
 
 
+def default_bytes_adapter(obj):
+    """Convert bytes objects to strings for json error renderer."""
+    if isinstance(obj, bytes):
+        return obj.decode('utf8')
+    return obj
+
+
 class _JSONError(exc.HTTPError):
-    def __init__(self, errors, status=400):
+    def __init__(self, renderer, renderer_args, errors, status=400):
         body = {'status': 'error', 'errors': errors}
-        Response.__init__(self, simplejson.dumps(body, use_decimal=True))
+        Response.__init__(self, renderer(body, **renderer_args))
         self.status = status
         self.content_type = 'application/json'
 
 
-def json_error(request):
-    """Returns an HTTPError with the given status and message.
+def json_error_handler(renderer, renderer_args=None):
```

Comment:
should maybe be called `json_error_handler_factory` ?

Reply:
Definitely, I'll update this once we figure out what to do with error handlers, as for the comment above.